# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1376426722 14400
#      Tue Aug 13 16:45:22 2013 -0400
# Node ID 6a6e3f1a824c0836428765515eb405112fe96d89
# Parent  96a1c132e3c6a9b0577acb1ef431fa426da8adc3
save history on exec (bug #39734)

* syscalls.cc (Fexec): Save history before exec.

diff --git a/libinterp/corefcn/syscalls.cc b/libinterp/corefcn/syscalls.cc
--- a/libinterp/corefcn/syscalls.cc
+++ b/libinterp/corefcn/syscalls.cc
@@ -33,26 +33,28 @@ along with Octave; see the file COPYING.
 #include <cstdio>
 #include <cstring>
 
 #include <sys/types.h>
 #include <unistd.h>
 
 #include <fcntl.h>
 
+#include "cmd-hist.h"
 #include "file-ops.h"
 #include "file-stat.h"
 #include "oct-env.h"
 #include "oct-syscalls.h"
 #include "oct-uname.h"
 
 #include "defun.h"
 #include "error.h"
 #include "gripes.h"
 #include "lo-utils.h"
+#include "oct-hist.h"
 #include "oct-map.h"
 #include "oct-obj.h"
 #include "oct-stdstrm.h"
 #include "oct-stream.h"
 #include "sysdep.h"
 #include "utils.h"
 #include "variables.h"
 #include "input.h"
@@ -215,16 +217,21 @@ error message.\n\
             {
               exec_args.resize (1);
 
               exec_args[0] = exec_file;
             }
 
           if (! error_state)
             {
+              octave_history_write_timestamp ();
+
+              if (! command_history::ignoring_entries ())
+                command_history::clean_up_and_save ();
+
               std::string msg;
 
               int status = octave_syscalls::execvp (exec_file, exec_args, msg);
 
               retval(1) = msg;
               retval(0) = status;
             }
         }
